client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java

```diff
@@ -119,6 +123,92 @@ public void onFailure(Exception e) {
 
     }
 
+    public void testGetLifecyclePolicy() throws IOException {
+        RestHighLevelClient client = highLevelClient();
+
+        // Set up some policies so we have something to get
+        {
+            Map<String, Phase> phases = new HashMap<>();
+            Map<String, LifecycleAction> hotActions = new HashMap<>();
+            hotActions.put(RolloverAction.NAME, new RolloverAction(
+                new ByteSizeValue(50, ByteSizeUnit.GB), null, null));
+            phases.put("hot", new Phase("hot", TimeValue.ZERO, hotActions));
+
+            Map<String, LifecycleAction> deleteActions =
+                Collections.singletonMap(DeleteAction.NAME,
+                    new DeleteAction());
+            phases.put("delete",
+                new Phase("delete",
+                    new TimeValue(90, TimeUnit.DAYS), deleteActions));
+
+            LifecyclePolicy policy = new LifecyclePolicy("my_policy",
+                phases);
+            PutLifecyclePolicyRequest putRequest =
+                new PutLifecyclePolicyRequest(policy);
+            LifecyclePolicy policy2 = new LifecyclePolicy("other_policy",
+                phases);
+            PutLifecyclePolicyRequest putRequest2 =
+                new PutLifecyclePolicyRequest(policy2);
+
+            AcknowledgedResponse putResponse = client.indexLifecycle().
+                putLifecyclePolicy(putRequest, RequestOptions.DEFAULT);
+            assertTrue(putResponse.isAcknowledged());
+            AcknowledgedResponse putResponse2 = client.indexLifecycle().
+                putLifecyclePolicy(putRequest2, RequestOptions.DEFAULT);
+            assertTrue(putResponse2.isAcknowledged());
+        }
+
+        // tag::ilm-get-lifecycle-policy-request
+        GetLifecyclePolicyRequest allRequest =
+            new GetLifecyclePolicyRequest(); // <1>
+        GetLifecyclePolicyRequest request =
+            new GetLifecyclePolicyRequest("my_policy", "other_policy"); // <2>
+        // end::ilm-get-lifecycle-policy-request
+
+        // tag::ilm-get-lifecycle-policy-execute
+        GetLifecyclePolicyResponse response = client.indexLifecycle()
+            .getLifecyclePolicy(request, RequestOptions.DEFAULT);
+        // end::ilm-get-lifecycle-policy-execute
+
+        // tag::ilm-get-lifecycle-policy-response
+        ImmutableOpenMap<String, LifecyclePolicyMetadata> policies =
+            response.getPolicies();
+        LifecyclePolicyMetadata myPolicyMetadata =
+            policies.get("my_policy"); // <1>
+        String myPolicyName = myPolicyMetadata.getName();
+        long version = myPolicyMetadata.getVersion();
+        String lastModified = myPolicyMetadata.getModifiedDateString();
+        long lastModifiedDate = myPolicyMetadata.getModifiedDate();
+        LifecyclePolicy myPolicy = myPolicyMetadata.getPolicy(); // <2>
+        // end::ilm-get-lifecycle-policy-response
```

Comment:
Can we add some asserts to ensure we have retrieved the correct thing from the API. Like checking that the policy we got back matches the policy we sent?